pkg/apis/intevents/v1alpha1/pullsubscription_validation_test.go

```diff
@@ -311,6 +340,23 @@ func TestPubSubCheckImmutableFields(t *testing.T) {
 			},
 			allowed: false,
 		},
+		"ServiceAccountName changed": {
```

Comment:
@nachocano Same as the topic_validation UT, I found we didn't check ServiceAccount is immutable here, so I added it for all versions.

Reply:
sg, thanks!